<a href="https://colab.research.google.com/github/iam-pattan/DL-lab/blob/main/Lab_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## LAB-2
New_dataset

In [ ]:
# !wget 'https://archive.ics.uci.edu/ml/machine-learning-databases/00445/Absenteeism_at_work_AAA.zip'

In [ ]:
# !unzip '/content/Absenteeism_at_work_AAA.zip'

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd
df = pd.read_excel('/content/drive/MyDrive/Colab Notebooks/Absenteeism_at_work.xls')
df.head()

ID  Reason for absence  ...  Body mass index  Absenteeism time in hours
0  11                  26  ...               30                          4
1  36                   0  ...               31                          0
2   3                  23  ...               31                          2
3   7                   7  ...               24                          4
4  11                  23  ...               30                          2

[5 rows x 21 columns]

In [ ]:
X = df.iloc[:,1:20]
y = df.iloc[:,20:].values

In [ ]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.25, random_state=42)

In [ ]:
X_train.shape, y_train.shape, X_test.shape, y_test.shape
# X_train.isnull().sum()
# X_train.values.shape

((555, 19), (555, 1), (185, 19), (185, 1))

In [ ]:
from keras.models import Sequential
from keras.layers import Dense
from keras.wrappers.scikit_learn import KerasRegressor
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold

In [ ]:
# create model
model = Sequential()
model.add(Dense(128, input_dim=19, kernel_initializer='normal', activation='relu'))
model.add(Dense(1, kernel_initializer='normal'))
# Compile model
model.compile(loss='mean_squared_error', optimizer='adam')

In [ ]:
model.fit(X_train, y_train,epochs= 20)

Epoch 1/20
18/18 [==============================] - 0s 1ms/step - loss: 12700843.9474
Epoch 2/20
18/18 [==============================] - 0s 1ms/step - loss: 1245299.1447
Epoch 3/20
18/18 [==============================] - 0s 1ms/step - loss: 93292.8604
Epoch 4/20
18/18 [==============================] - 0s 1ms/step - loss: 31161.1980
Epoch 5/20
18/18 [==============================] - 0s 1ms/step - loss: 5556.4310
Epoch 6/20
18/18 [==============================] - 0s 1ms/step - loss: 1081.1389
Epoch 7/20
18/18 [==============================] - 0s 1ms/step - loss: 328.1655
Epoch 8/20
18/18 [==============================] - 0s 1ms/step - loss: 255.6362
Epoch 9/20
18/18 [==============================] - 0s 1ms/step - loss: 178.6077
Epoch 10/20
18/18 [==============================] - 0s 1ms/step - loss: 200.6202
Epoch 11/20
18/18 [==============================] - 0s 1ms/step - loss: 172.2204
Epoch 12/20
18/18 [==============================] - 0s 1ms/step - loss: 183.2415
Epoch 13/2

In [ ]:
model.evaluate(X_test,y_test)

6/6 [==============================] - 0s 2ms/step - loss: 96.3411


96.34107208251953

In [ ]:
from sklearn.svm import SVR

svr_regressor = SVR(kernel='rbf')
svr_regressor.fit(X,y.ravel())

SVR(C=1.0, cache_size=200, coef0=0.0, degree=3, epsilon=0.1, gamma='scale',
    kernel='rbf', max_iter=-1, shrinking=True, tol=0.001, verbose=False)

In [ ]:
# from sklearn.preprocessing import StandardScaler, MinMaxScaler

# # scaler = StandardScaler()
# sclaer = MinMaxScaler()
# dataframe1 = scaler.fit(df)
# dataframe2 = scaler.transform(df)
# dataframe2

In [ ]:
prediction = svr_regressor.predict(X)

In [ ]:
y_pred = svr_regressor.predict(X_test)

In [ ]:
from sklearn.metrics import mean_squared_error, explained_variance_score

print('MSE:{}'.format(mean_squared_error(y_test,y_pred)))
print('Explained variance score:{}'.format(explained_variance_score(y_test,y_pred)))

MSE:104.8086025638058
Explained variance score:0.008020669732768382


Achieved validation accuracy of `96.34%` on Absenteeism dataset.

Input size of 19 columns are considered to predict the employees attendance using simple Perceptron.

Tried using SVR.